configure.ac

```diff
@@ -0,0 +1,100 @@
+
+AC_INIT([kvazaar], m4_esyscmd([printf $(awk '/#define KVZ_VERSION/ { print $3 }' src/global.h)]))
+AC_CONFIG_SRCDIR([src/encmain.c])
+
+# not used, but it prevents configure from adding a lot of defines to the CFLAGS
+AC_CONFIG_HEADERS([config.h])
+
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR([build-aux])
+
+AM_INIT_AUTOMAKE([-Wall -Werror dist-bzip2 dist-xz foreign subdir-objects])
+AM_SILENT_RULES([yes])
+
+AC_PROG_CC
+AM_PROG_AR
+
+LT_INIT([win32-dll])
+
+
+CFLAGS="-Wall -Wtype-limits -Wvla -I$srcdir/src -ftree-vectorize -fvisibility=hidden -std=gnu99 -DKVZ_DLL_EXPORTS $CFLAGS"
+ASFLAGS="-I$srcdir/src/extras $ASFLAGS"
+LIBS="-lm $LIBS"
+
+
+# --disable-werror
+AC_ARG_ENABLE([werror], [AS_HELP_STRING([--disable-werror], [don't treat warnings as errors [no]])],
+              [], [CFLAGS="-Werror $CFLAGS"]
+)
+
+
+# check for getopt
+AC_CHECK_HEADER([getopt.h], [], [CFLAGS="$CFLAGS -I$srcdir/src/extras"])
+
+
+# host and cpu specific settings
+AS_CASE([$host_cpu],
+        [i?86], [BITS="32" ASFLAGS="$ASFLAGS -DARCH_X86_64=0" X86="true"],
+        [x86_64], [BITS="64" ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -m amd64" X86="true"],
+        [powerpc*], [PPC="true"],
+        [arm*], [ARM="true"]
+)
+
+AS_CASE([$host_os],
+        [darwin*], [
+         ASFLAGS="$ASFLAGS -f macho$BITS -DPREFIX"
+        ],
+        [cygwin*|msys*|mingw*], [
+         CFLAGS="$CFLAGS -D__USE_MINGW_ANSI_STDIO=1"
+         AS_IF(
+               [test "x$BITS" = "x32"], [
+                ASFLAGS="$ASFLAGS -fwin32 -DPREFIX -DHAVE_ALIGNED_STACK=0"
+               ], [
+                ASFLAGS="$ASFLAGS -fwin64 -DHAVE_ALIGNED_STACK=1"
+               ]
+         )
+        ], [
+         ASFLAGS="$ASFLAGS -f elf$BITS"
+         CFLAGS="$CFLAGS -pthread"
+         LDFLAGS="$LDFLAGS -pthread -Wl,-z,noexecstack"
+         LIBS="$LIBS -lpthread"
```

Comment:
This happened with each of gcc 4.6.4, gcc 4.9.2 and clang 3.0 on Ubuntu 12.04 x86_64.